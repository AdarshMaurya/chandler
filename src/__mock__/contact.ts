import {IContact} from '@models';

export const contact1: IContact = {
  id: 2,
  updated_at: '2018-12-17',
  addresses: [
    {
      street: '5815 Park',
      city: 'Montreal',
    },
  ],
  first_name: 'Theo',
  last_name: 'Mathieu',
  information: {
    relationships: {
      family: {
        total: 0,
        contacts: [],
      },
      love: {
        total: 0,
        contacts: [],
      },
      work: {
        total: 0,
        contacts: [],
      },
      friend: {
        total: 0,
        contacts: [],
      },
    },
    how_you_met: {
      first_met_date: {
        date: '1991-12-17',
      },
      first_met_through_contact: null,
      general_information: 'test',
    },
    avatar: {
      default_avatar_color: 'blue',
      url: 'testurl',
      source: 'external',
    },
    career: {
      company: 'Google',
      job: 'Developer',
      linkedin_profile_url: 'http://mylinkedin',
    },
    dates: {
      birthdate: {
        date: null,
      },
      deceased_date: {
        date: null,
      },
    },
  },
  statistics: {
    number_of_activities: 2,
    number_of_calls: 3,
    number_of_debts: 4,
    number_of_gifts: 5,
    number_of_notes: 6,
    number_of_reminders: 9,
    number_of_tasks: 1,
  },
  contactFields: [
    {
      contact_field_type: {
        fontawesome_icon: 'fa fa-facebook',
        name: 'name',
        protocol: 'http',
        type: 'email',
      },
      content: 'test@gmail.com',
    },
  ],
};
