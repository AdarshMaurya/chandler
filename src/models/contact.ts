interface IContactRelationships {
  family: {
    total: number;
    contacts: IRelationship[];
  };
  friend: {
    total: number;
    contacts: IRelationship[];
  };
  love: {
    total: number;
    contacts: IRelationship[];
  };
  work: {
    total: number;
    contacts: IRelationship[];
  };
}

interface IRelationship {
  contact: IContact;
  relationship: {
    id: string;
    name: string;
  };
}

interface IAgeDate {
  is_age_based?: boolean;
  is_year_unknown?: boolean;
  date: string;
}

interface IContactInformation {
  career: {
    job?: string;
    company?: string;
    linkedin_profile_url?: string;
  };
  avatar: {
    default_avatar_color: string;
    url: string;
    source: 'external' | 'internal';
  };
  relationships: IContactRelationships;
  dates: {
    birthdate: IAgeDate;
    deceased_date: IAgeDate;
  };
  how_you_met: {
    first_met_date: IAgeDate;
    first_met_through_contact: IContact;
    general_information: string;
  };
}

export interface IContact {
  id: number;
  first_name: string;
  last_name?: string;
  information: IContactInformation;
  updated_at: string;
  statistics: {
    number_of_calls: number;
    number_of_notes: number;
    number_of_activities: number;
    number_of_reminders: number;
    number_of_tasks: number;
    number_of_gifts: number;
    number_of_debts: number;
  };

  reminders?: number[];
  gifts?: number[];
  activities?: number[];
  tasks?: number[];
  calls?: number[];
  debts?: number[];
  notes?: number[];
}
