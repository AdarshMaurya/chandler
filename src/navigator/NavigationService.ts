import {NavigationActions, StackActions} from 'react-navigation';

let navigator;

export const setTopLevelNavigator = navigatorRef => {
  navigator = navigatorRef;
};

export const navigate = (routeName: string, params?: any) => {
  navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
};

export const pop = () => {
  navigator.dispatch(StackActions.pop());
};

export type INavigateAction = typeof navigate;
export type IPopAction = typeof pop;
