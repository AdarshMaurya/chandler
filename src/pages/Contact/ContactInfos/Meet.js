import React, {Component} from 'react';
import {Text} from 'react-native';

export class Meet extends Component {
  render() {
    return <Text>Meet</Text>;
  }
}
