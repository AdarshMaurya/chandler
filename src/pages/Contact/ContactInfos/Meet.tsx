import {IContact} from '@models';
import moment from 'moment';
import React, {PureComponent} from 'react';
import {Text, View} from 'react-native';

import {ListItem} from '@components';
import {I18n} from '@i18n';
import {getName} from '@utils/contacts';
import {styles} from './Meet.styles';

interface IMeetProps {
  contact: IContact;
}

export class Meet extends PureComponent<IMeetProps, {}> {
  public render() {
    const {contact} = this.props;
    const labels = this.getLabels(contact);

    return (
      <View style={styles.container}>
        {labels.length === 0 && (
          <Text style={styles.noInformation}>
            {I18n.t('contacts:meet.none')}
          </Text>
        )}
        {labels.map((label, index) => (
          <ListItem
            key={index}
            title={label}
            lastItem={labels.length - 1 === index}
          />
        ))}
      </View>
    );
  }

  private getLabels(contact: IContact) {
    const res = [];

    // date
    const date = contact.information.how_you_met.first_met_date.date;
    if (date) {
      res.push(moment(date).calendar());
    }

    // met through
    const metThrough =
      contact.information.how_you_met.first_met_through_contact;
    if (metThrough) {
      res.push(I18n.t('contacts:meet.through', {name: getName(metThrough)}));
    }

    // description
    const description = contact.information.how_you_met.general_information;
    if (description) {
      res.push(description);
    }

    return res;
  }
}
