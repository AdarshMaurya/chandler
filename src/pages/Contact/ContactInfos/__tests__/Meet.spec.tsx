import {navigate} from '@navigator/NavigationService';
import {contact1} from '@src/__mock__/contact';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {Meet} from '../Meet';

const navigateMock = navigate as jest.Mock;

describe('Pages', () => {
  describe('Contact / Meet', () => {
    const contact = {
      first_name: 'Theo',
      last_name: 'Mathieu',
    };

    it('should renders correctly', () => {
      expect(toJson(shallow(<Meet contact={contact1} />))).toMatchSnapshot();
    });
  });
});
