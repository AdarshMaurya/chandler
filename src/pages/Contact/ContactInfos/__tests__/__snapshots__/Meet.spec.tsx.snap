// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Contact / Meet should renders correctly 1`] = `
<Component
  style={
    Object {
      "marginTop": 5,
    }
  }
>
  <ListItem
    key="0"
    lastItem={false}
    title="12/17/1991"
  />
  <ListItem
    key="1"
    lastItem={true}
    title="test"
  />
</Component>
`;
